client/http/types.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2023 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package http
+
+import "time"
+
+// NOTICE: the structures below are copied from the PD API definitions.
+// Please make sure the consistency if any change happens to the PD API.
+
+// RegionInfo stores the information of one region.
+type RegionInfo struct {
+	ID              int64            `json:"id"`
+	StartKey        string           `json:"start_key"`
+	EndKey          string           `json:"end_key"`
+	Epoch           RegionEpoch      `json:"epoch"`
+	Peers           []RegionPeer     `json:"peers"`
+	Leader          RegionPeer       `json:"leader"`
+	DownPeers       []RegionPeerStat `json:"down_peers"`
+	PendingPeers    []RegionPeer     `json:"pending_peers"`
+	WrittenBytes    uint64           `json:"written_bytes"`
+	ReadBytes       uint64           `json:"read_bytes"`
+	ApproximateSize int64            `json:"approximate_size"`
+	ApproximateKeys int64            `json:"approximate_keys"`
+
+	ReplicationStatus *ReplicationStatus `json:"replication_status,omitempty"`
+}
+
+// GetStartKey gets the start key of the region.
+func (r *RegionInfo) GetStartKey() string { return r.StartKey }
+
+// GetEndKey gets the end key of the region.
+func (r *RegionInfo) GetEndKey() string { return r.EndKey }
+
+// RegionEpoch stores the information about its epoch.
+type RegionEpoch struct {
+	ConfVer int64 `json:"conf_ver"`
+	Version int64 `json:"version"`
+}
+
+// RegionPeer stores information of one peer.
+type RegionPeer struct {
+	ID        int64 `json:"id"`
+	StoreID   int64 `json:"store_id"`
+	IsLearner bool  `json:"is_learner"`
+}
+
+// RegionPeerStat stores one field `DownSec` which indicates how long it's down than `RegionPeer`.
+type RegionPeerStat struct {
+	Peer    RegionPeer `json:"peer"`
+	DownSec int64      `json:"down_seconds"`
+}
+
+// ReplicationStatus represents the replication mode status of the region.
+type ReplicationStatus struct {
+	State   string `json:"state"`
+	StateID int64  `json:"state_id"`
+}
+
+// RegionsInfo stores the information of regions.
+type RegionsInfo struct {
+	Count   int64        `json:"count"`
+	Regions []RegionInfo `json:"regions"`
+}
+
+// Merge merges two RegionsInfo together and returns a new one.
+func (ri *RegionsInfo) Merge(other *RegionsInfo) *RegionsInfo {
```

Comment:
Where it is used?

Reply:
After checking, I found that it was not used. Removed.

Reply:
It will be used here:

```shell
pkg/executor/infoschema_reader.go:1609:37: allRegionsInfo.Merge undefined (type *"github.com/tikv/pd/client/http".RegionsInfo has no field or method Merge)
```

Reply:
https://github.com/pingcap/tidb/blob/6e8df186f51b90d57e29875b03afe317298bc949/pkg/executor/infoschema_reader.go#L1609

Reply:
It's better to add a unit test for this function